### DISCUSSION TITLE
Running openHASP over Ethernet or PoE

### DISCUSSION THREAD

**Discussion by lilmuckers** [HASwitchPlate/openHASP#430] [Hardware]
I'm designing a pretty extensive re-wire of my house, and i'm looking to install a bunch of sensors, devices, speakers, rhasspy node and openHASP plates in each room.
I don't trust wifi in my house at all due to big thick walls and a flagrantly unreliable ISP, so i'm investing in using PoE and ZigBee as much as possible in order to reduce noise on the network, and mitigate from any wifi-based flakiness on the devices themselves in order to achieve the all important Partner Approval Factor.

I'm wiring in PoE network for my security system and NVR, as well as for using esp32 rhasspy - which streams audio over the network so i want it to be PoE/ethernet based to avoid noisy networks.

Anyway, that's only context.
I would like to use PoE to provide power and network to the plates. In my researches i have looked at the ESP32-POE and ESP32-POE-ISO as potential candidates with the screens.

But mostly what i'm looking for is any advice about what i may be (probably naively) getting wrong and what i might need to do to enable ethernet rather than wifi networking. Even a place to start with modifying the code.

**Comment by fvanroie** [comment]
There is the `lanbon_l8_eth.ini`and `ttgo-esp32-poe_ili9341.ini` that have ehternet ports. I would suggest starting with those.
Both [configurations](https://github.com/HASwitchPlate/openHASP/tree/master/user_setups/esp32) are _not_ actively maintained, so they will need testing and quite possibly some work to get going (again).

I don't own the ESP32-POE or ESP32-POE-ISO so I can't say much about these...
The [Lilygo®Ttgo T-Internet-Poe ESP32](https://s.click.aliexpress.com/e/_Dl4LdSD) board has been tested and it did work.
It was a while back though, so the current version needs testing and some work for sure to bring it up-to-date.

My gripe with most POE boards is that they have minimal flash and no PSram. It's okay for basic UIs, but custom fonts and images will be an issue. A notable exception being the [POE-FeathingWing](https://www.tindie.com/products/silicognition/poe-featherwing/) that could *theoretically* be paired with a more capable ESP32. _(not tested)_

The [Lanbon L8 MOD](https://github.com/HASwitchPlate/openHASP/discussions/65?sort=new#discussioncomment-2928054) also uses an Ethernet SPI adapter, but that [might be slow](https://github.com/HASwitchPlate/openHASP/issues/381).

What display and size are you planning to use? Ehternet takes up a fair amount of pins, so you are limited to SPI displays.

I don't see a reason why it can't work. There just hasn't been much demand for POE or any users that can test/maintain the configs.
If you're able to do some testing and troubleshooting on the existing configs and/or tweak them for the Olimex boards that would be great.

**Comment by fantasmisiciliani** [comment]
I tested the SPI ethernet (W5500) on a WT32-SC01 Plus and it works with no need of any hardware mod on the main board. I just soldered a small LDO board (AMS1117) on the W5500 Lite to get 3.3V from the 5V pin of the WT32 I/O connector.
I haven't done any speed test, but pages from the web interfaces are loaded pretty fast and all works as expected, except for GPIO outputs in inverted mode that cannot be turned off after switched on. I don't think this is related with Ethernet and I'm going to open a bug after having browsed all the discussions.
To compile the firmware, you have to add some build flag and 3 files from the ESP-IDF, as I described in the issue #381, but nothing of impossible.
Then, you can use an active POE splitter with USB type C to power the board and connect it to your LAN.
![IMG_20230402_171518_resized_20230402_051643198](https://user-images.githubusercontent.com/77593353/229581116-0ec13fce-6a4f-461b-8c0f-6ed8b1231bc8.jpg)
